src/Util.cpp

```diff
@@ -1308,6 +1312,8 @@ to_lowercase(string_view string)
 void
 traverse(const std::string& path, const TraverseVisitor& visitor)
 {
+  // Prefer POSIX opendir()/readdir() implementation for performance reasons.
```

Comment:
Since the implementations share very little I think that it would make sense to write them as separate implementations, i.e.:

```
#ifdef HAVE_DIRENT_H

void
traverse(const std::string& path, const TraverseVisitor& visitor)
{
  // opendir...
}

#else // If not available, use C++17 std::filesystem implementation.

void
traverse(const std::string& path, const TraverseVisitor& visitor)
{
  // std::filesystem...
}

#endif
```

Then we can also avoid the clang-format workaround.

Reply:
This is done, and it did in fact fix the clang-format issue. This was the last issue in your list.